setup.py

```diff
@@ -443,7 +444,8 @@ def run_install_headers(self):
 
     if "freetype" in e.name and sys.platform not in ("darwin", "win32"):
         # TODO: fix freetype issues here
-        e.extra_compile_args.append("-Wno-error=unused-but-set-variable")
+        if sysconfig.get_config_var("MAINCC") != "clang":        
```

Comment:
I'm not sure how robust this check is...

I had added the `sys.platform not in ("darwin", "win32")` check above just to block this arg being set on clang and MSVC respectively (this flag should only be set on gcc) but this is not a robust check either

The best solution to this that I can find involves more distutils hacks (https://stackoverflow.com/a/1671060) but it should give the accurate compiler name that is being used to compile the code